core/src/main/java/org/apache/james/mime4j/util/MimeParameterMapping.java

```diff
@@ -36,13 +41,41 @@ public String get(String name) {
         return parameters.get(name);
     }
 
-    public void addParameter(String name, String value) {
-        String key = removeSectionFromName(name).toLowerCase();
+    public void addParameter(final String name, String value) {
+        String key = name;
+        int sectionDelimiter = name.indexOf("*");
+        if (sectionDelimiter > -1) {
+            key = key.substring(0, sectionDelimiter);
+        }
+        key = key.toLowerCase(Locale.ROOT);
+        //switch on whether there's a section in the key
+        if (sectionDelimiter < 0) {
+            handleNoSection(key, value);
+        } else {
+            handleSection(key, value);
+        }
+    }
+
+    private void handleSection(String key, String value) {
+        //if the key there was not sectioned, remove it.
```

Comment:
Maybe we could add that `we conform to https://bugzilla.mozilla.org/show_bug.cgi?id=588781 as part of MIME4J-332` so that the decision on the edge case, which is not obvious actually gets documented in the code too ?